ESP8266Interface.h

```diff
@@ -354,7 +354,7 @@ class ESP8266Interface : public NetworkStack, public WiFiInterface {
     // Drivers's socket info
     struct _sock_info {
         bool open;
-        uint16_t sport;
+        int32_t sport;
```

Comment:
No point expanding this to hold -1, when -1 has no special meaning to you. You just happen to store -1 when closed, but it has no meaning, as far as I can tell. You could just store 0.